cdap-docs/admin-manual/source/operations/logging.rst

```diff
@@ -1,77 +1,573 @@
+for log processing pipeline configurations. Each pipeline is defined by a file in the
+Logback XML format, with ``.xml`` as the file name extension.
+
+These :ref:`properties <appendix-cdap-default-logging>` control the CDAP log pipeline:
+
+.. generated by code in conf.py for this manual
+.. include:: /../target/_includes/logging-pipeline-configuration-1.rst
+
+.. _logging-monitoring-cdap-site-xml-configuration:
+
+These :ref:`properties <appendix-cdap-default-logging>` control both the CDAP log pipeline
+and custom log pipelines:
+
+.. generated by code in conf.py for this manual
+.. include:: /../target/_includes/logging-pipeline-configuration-2.rst
+
+The ``log.process.pipeline.*`` properties can be over-ridden and specified at the
+custom pipeline level by providing a value in a pipeline's "logback" file for any of
+these properties.
+
+
+Logging Framework
+=================
+
+.. highlight:: console
+
+This diagram shows in greater detail the components and steps CDAP follows when logging
+programs of an application and system services with the logging framework:
+
+  - Logs are collected from individual programs running in YARN containers.
+
+  - YARN writes the log messages emitted by containers to files inside the container.
+
+  - CDAP System Services run (depending on the service) either on cluster edge nodes or in
+    YARN containers. Where they run determines the file that configures that service's
+    logging.
+
+  - In addition, CDAP programs publish these messages to Kafka.
+
+  - The Log Saver Service (*Log.saver*) is configured to read log messages for the
+    ``logs.user-v2`` Kafka topic (set by the property ``log.kafka.topic``). The number of
+    log saver instances can be scaled to process the Kafka partitions in parallel, if
+    needed.
+    
+    *Log.saver*, by default, runs only the CDAP Log Pipeline: it reads the messages from
+    Kafka, groups them by program or application, buffers and sorts them in memory, and
+    finally persists them to files in HDFS. Each of these files corresponds to one program
+    or application, depending on how the grouping is configured. (This is set by the
+    property :ref:`log.publish.partition.key <logging-monitoring-writing-logs-to-kafka>`,
+    described :ref:`below <logging-monitoring-configuring-the-log-saver-service>`.)
+
+    *Note:* These files are configured to rotate based on time and size; they can be
+    changed using the properties ``log.pipeline.cdap.file.max.size.bytes`` and
+    ``log.pipeline.cdap.file.max.lifetime.ms`` in the :ref:`cdap-site.xml
+    <appendix-cdap-default-logging>` file as described in 
+    :ref:`logging-monitoring-log-pipeline-configuration`.
+
+  - In addition, custom log pipelines can be configured by adding an XML file in a
+    prescribed location. Each pipeline buffers log messages in memory and sorts them based
+    on their timestamp. 
+   
+  - In addition to persisting logs to files, the Log.saver also emits metrics about the
```

Comment:
Do we document anywhere what the names of these metrics are? If so, a link would help here.